graylog2-web-interface/src/views/pages/NewSearchPage.test.jsx

```diff
@@ -0,0 +1,194 @@
+// @flow strict
+import * as React from 'react';
+import { render, cleanup, wait, waitForElement, fireEvent } from 'wrappedTestingLibrary';
+
+import { StoreMock as MockStore } from 'helpers/mocking';
+import asMock from 'helpers/mocking/AsMock';
+
+import { processHooks } from 'views/logic/views/ViewLoader';
+import { syncWithQueryParameters } from 'views/hooks/SyncWithQueryParameters';
+import { ViewActions } from 'views/stores/ViewStore';
+import { ViewManagementActions } from 'views/stores/ViewManagementStore';
+import NewViewLoaderContext from 'views/logic/NewViewLoaderContext';
+import Search from 'views/logic/search/Search';
+import View from 'views/logic/views/View';
+import ViewLoaderContext from 'views/logic/ViewLoaderContext';
+
+import NewSearchPage from './NewSearchPage';
+
+const mockExtendedSearchPage = jest.fn(() => <div>Extended search page</div>);
+const mockView = View.create()
+  .toBuilder()
+  .type(View.Type.Search)
+  .search(Search.builder().build())
+  .build();
+
+jest.mock('react-router', () => ({ withRouter: x => x }));
+jest.mock('./ExtendedSearchPage', () => mockExtendedSearchPage);
+jest.mock('views/stores/SearchStore', () => MockStore());
+jest.mock('views/stores/ViewStore', () => ({
+  ViewActions: { create: jest.fn(() => Promise.resolve({ view: mockView })) },
+}));
+jest.mock('views/hooks/SyncWithQueryParameters', () => ({
+  syncWithQueryParameters: jest.fn(),
+}));
+jest.mock('views/stores/ViewManagementStore', () => ({
+  ViewManagementActions: {
+    get: jest.fn(() => Promise.resolve()),
+  },
+}));
+jest.mock('views/logic/views/ViewLoader', () => {
+  const originalModule = jest.requireActual('views/logic/views/ViewLoader');
+  return {
+    __esModule: true,
+    ...originalModule,
+    processHooks: jest.fn((promise, loadHooks, executeHooks, query, onSuccess) => Promise.resolve().then(onSuccess)),
+  };
+});
```

Comment:
Question: So far we are not using `requireActual`. it looked like the best solution for me, in the context of the following case:
This is the related import `import ViewLoader, { processHooks } from 'views/logic/views/ViewLoader';`
We need to mock `processHooks`, to test if it got called correctly. But I don't want to mock the ViewLoader, because I need to trigger e.g. the onError function we are passing to the `ViewLoader`. This is the main reason for the above implementation. 

Having a look at e.g. the the `ShowViewPage`, I saw a different usage of the `ViewLoader` regarding the import. The `ViewLoader` and `processHooks` are getting imported the same way, but we are not using the `ViewLoader` directly. We are defining it as a default prop. 

My question is, should we unify the usage of the `ViewLoader` for `NewDashoardPage`, `NewSearchPage`, `ShowViewPage` and `StreamSearchPage`?
The Implementation with the default prop would simplify the described test case.